File: nomad-samples/js/content-admin-api/content/add-tag-or-collection.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function addTagOrCollection(AUTH_TOKEN, TYPE, CONTENT_ID, TAG_ID, CONTENT_DEFINITION, TAG_NAME, CREATE_NEW) 
{
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);

    const BODY = {
        items: [
            {
                contentDefinition: CONTENT_DEFINITION,
                contentId: CONTENT_ID,
                name: TAG_NAME,
                createNew: CREATE_NEW
            }
        ]
    };

    if (TAG_ID)
    {
        BODY.items[0].tagId = TAG_ID;
    }

    // Post
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/admin/${TYPE}/content`, {
        method: "POST",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    }).catch((exception) => {
        throw exception;
    });

    console.log(JSON.stringify(BODY, null, 4));

    // Check for success
    if (RESPONSE.ok) {
        // Get the JSON from the response
        const INFO = await RESPONSE.json();

        return INFO;
    }
		
  	// There was an error
    await apiExceptionHandler(RESPONSE, "Add tag or collection failed");
}

File: nomad-samples/python/movies/content_admin_api/content/add_tag_or_collection.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def add_tag_or_collection(AUTH_TOKEN: str, TYPE: str, CONTENT_ID: str, CONTENT_DEFINITION: str, NAME: str, TAG_ID: str, CREATE_NEW: bool) -> dict:

    # Check for valid parameters
    if (not AUTH_TOKEN):
        raise Exception("Authentication Token: The authentication token is invalid")

    API_URL = ADMIN_API_URL + "/admin/" + TYPE + "/content"
        
    # Create header for the request
    HEADERS = {
  	    "Authorization": "Bearer " + AUTH_TOKEN,
        "Content-Type": "application/json"
    }

    # Build the payload BODY
    BODY = {
        "items": [
            {
                "contentDefinition": CONTENT_DEFINITION,
                "contentId": CONTENT_ID,
                "name": NAME,
                "createNew": CREATE_NEW
            }
        ]
    }

    if TAG_ID != "":
        BODY["items"]["tagId"] = TAG_ID

    try:
        # Send the request
        RESPONSE = requests.post(API_URL, headers= HEADERS, data= json.dumps(BODY))

        if not RESPONSE.ok:
            raise Exception("Response returned " + str(RESPONSE.status_code))

        return json.loads(RESPONSE.text)

    except:
        api_exception_handler(RESPONSE, "add tag or collection failed")

